### DISCUSSION TITLE
Adding a new batch system to Columnflow

### DISCUSSION THREAD

**Discussion by gsaha009** [columnflow/columnflow#440] [General]
Hello CF experts,

I am going to use a batch system which is not included in columnflow (/law?). In order to include that, would anyone kindly provide any ideas?
 
I was going through the HTCondor implemetation and what I have understood the main changes would be are the following:
 - A new dir in [contrib](https://github.com/riga/law/tree/master/law/contrib): `pbs`
 - Define `config`, `job`, `util`, `workflow` and `__init__` there
 - Write a `remoteWorkFlow` in [remote.py](https://github.com/columnflow/columnflow/blob/master/columnflow/tasks/framework/remote.py)
 - A few more changes

Looks like a lot to change though. Any sort of comments/ideas would be very helpful.
Thank you :)

Gourab

**Comment by pkausw** [comment]
Hi @gsaha009 ,

indeed, what you describe above looks correct to me at first glance. Few comments,

- in principle, you don't have to use the same file structure as was used for other workflows - it's python after all, and you can basically import anything from anywhere. That being said, using the same structure will enable you to recycle code much more efficiently and will make your code more comparable of course.
- You also don't need to implement anything into law/columnflow itself. In principle, you should be able to create a structure in you analysis (if your new workflow is analysis specific of course) and import the things you need from law (e.g. things from `law.job.base`) and columnflow (e.g. the `RemoteWorkflowMixin` class).

This seems like a nice potential addition, so I'm sure we'll be happy to give you feedback if you run into problems while implementing this. Also tagging @riga here since he implemented the other remote workflows.

Cheers!

**Reply by gsaha009** [reply]
Hi @pkausw @riga 
Thank you for this detailed explanation. I have not started working on it yet. I will do a kick start soon and let you know some obvious doubts and challenges that I am going to face.

**Comment by pkausw** [comment]
Hi @gsaha009 , 

Is this discussion still open or can we close it?

Cheers,
Philip

**Comment by gsaha009** [comment]
Hi @pkausw 
Let's keep this discussion open. I tried to include a new batch system after discussing with @riga, but it did not work then. I will have a look at it again.
Otherwise, you can close it. I will reopen it if necessary. 
Thanks
